### PR TITLE
Delete redundant code in RevivalShop and refactorize it

### PR DESCRIPTION
**Pull request by michalbie**
Deleted reduntant code by putting it in functions. Also move setting variables from _process to _ready and upgrade functions.

### CODE REVIEW

**Comment by michalbie** [comment]
Had some conflicts because i didnt rebased master since last pull request merge (fix_events), but should be working fine

File: Scenes/Screens/RevivalShop/RevivalShopItem.gd [hunk 1]
```diff
@@ -2,7 +2,25 @@ extends HBoxContainer
...
+export (String) var upgrade_func
+export (String) var button_sign
+var count_text
+
 
 func _ready():
 	$Info.set_text(info)
 	$Count.set("custom_colors/font_color", color)
+	
+func update():
```
Comment (Ajver) [inline Scenes/Screens/RevivalShop/RevivalShopItem.gd:15]: `update` method is already in Godot: [READ](https://docs.godotengine.org/en/3.2/tutorials/2d/custom_drawing_in_2d.html?#updating). Maybe `refresh`? `_update`? `_update_self`? I don't have got 'nice' idea now... But surely NOT just `update`.

File: Scenes/Screens/RevivalShop/RevivalShop.gd [hunk 2]
```diff
@@ -16,194 +16,112 @@ const BASIC_DAMAGE_PRCE = 0
...
-	set_basic_damage_button()
-	set_basic_damage_count()
-	set_basic_hp_button()
-	set_basic_hp_count()
-	set_items_price_button()
-	set_items_price_count()
+	#connect_buttons_signals()
+	send_count_text("Item_enemies_per_level", get_parent().find_node("LevelManager").dwarves_per_level)
+	send_count_text("Item_earn_gold", 'x' + String(GameData.additional_gold_multipler))
+	send_count_text("Item_earn_xp", 'x' + String(GameData.additional_xp_multipler))
```
Comment (B0ZIA) [inline Scenes/Screens/RevivalShop/RevivalShop.gd:25]: Now in game we don't have xp. You can hide the option Item_earn_xp in RevivalShop.tscn and if you want comment off these type lines.

**Comment by michalbie** [comment]
@Ajver @B0ZIA I resolved all conflicts but i don't know if it works properly. Something related to revival shop was merged into master and I had to delete some code (for example, because of doubled code in some places). Somebody needs to check if revival shop works before this pull request will be merged into master. Next time if somebody wants to add something related to revivalShop he should make a new branch from this (revivalShop_refactor) and work on it to avoid conflicts. But more frequent merging pull requests also would work ;)

**Comment by Ajver** [comment]
It's OK now @michalbie . I fixed some little issue (there were two old methods, that I deleted and pushed to this PR: https://github.com/Ajver/EvD/pull/83

**Pull request closed without merging**